bin/generate.py

```diff
@@ -1699,60 +1700,24 @@ def generate_dir_unlisted(d):
         self.update_gitignore_file()
 
     def update_gitignore_file(self):
-        gitignorefile = self.problem.path / 'data/.gitignore'
-
-        if not self.gitignore_generated:
-            # Remove existing .gitignore file if created by BAPCtools.
-            if gitignorefile.is_file():
-                text = gitignorefile.read_text()
-                if text.startswith('# GENERATED BY BAPCtools'):
-                    gitignorefile.unlink()
-                    log('Deleted data/.gitignore.')
-            return
-
-        # Collect all listed testcases
-        # and gitignore them in the data/ directory.
-        # Sample cases are never ignored.
-        # When only generating a subset of testcases, we also keep existing
-        # entries not matching the filter.
-        cases_to_ignore = []
-
-        def maybe_ignore_testcase(t):
-            if not (t.inline or t.sample):
-                cases_to_ignore.append(t.path)
+        gitignorefile = self.problem.path / '.gitignore'
 
-        self.root_dir.walk(maybe_ignore_testcase, None)
-
-        if config.args.testcases and gitignorefile.is_file():
-            # If there is an existing .gitignore and a list of testcases is
-            # passed, keep entries that are not processed in this run.
-            for line in gitignorefile.read_text().splitlines():
-                if line[0] == '#' or line == '.gitignore':
-                    continue
-                assert line.endswith('.*')
-                line = Path(line).with_suffix('')
-                if not process_testcase(self.problem, line):
-                    cases_to_ignore.append(line)
-        cases_to_ignore.sort()
-
-        if len(cases_to_ignore) == 0:
-            return
+        content = """#GENERATED BY BAPCtools
+data/
+!data/sample/
+"""
 
         if gitignorefile.is_file():
-            text = gitignorefile.read_text()
-            if not text.startswith('# GENERATED BY BAPCtools'):
-                warn('Not overwriting existing data/.gitignore file.')
-                return
+            # if there is any rule fore data/ we expect that the user knows
+            # what he does.
+            if gitignorefile.read_text().find('data/') == -1:
+                with gitignorefile.open("a") as f:
+                    f.write(content)
```

Comment:
Maybe write a preceding newline to not create bugs when `f` doesn't end in a newline.